avl_tree/AvlTree.java

```diff
@@ -0,0 +1,306 @@
+import java.util.Scanner;
+import java.util.NoSuchElementException;
+import java.util.InputMismatchException;
+
+public class AvlTree {
+
+	public static void main(String[] args) {
+		AVL avl = new AVL();
+		Scanner inp = new Scanner(System.in);
+		while (true) {

```

Comment:
Please hard-code the inputs. This essentially means not taking the input from the user as it consumes some time. Also user might not be sure of the format of input. Having hard-coded inputs in code solves these problems. See Stack.java if you need an example.